coordinator/internal/transitengine/transitengine_test.go

```diff
@@ -0,0 +1,164 @@
+		},
+		"f_name_with_forsl": {
+			Plaintext:      "vpIh",
+			name:           "wrong/url",
+			AssociatedData: "vpIh",
+			expStatus:      404,
+		},
+		"f_bad-request-nob64plain": {
+			Plaintext:      "thi$$hoU_ld+*notwork",
+			name:           "autounseal",
+			AssociatedData: "vpIh",
+			expStatus:      400,
+		},
+		"f_bad-request-nob64AD": {
+			Plaintext:      "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			name:           "autounseal",
+			AssociatedData: "thi$$hoU_ld+*notwork",
+			expStatus:      400,
+		},
+		"f_neg_version": {
+			Plaintext:      "lT3rQGMlxq680DdSKfIYYcfyCfMnP9ikxaO5b0mGRKRl4qNL3W9xkSW3QmaMwozCRfNMZhhDCbYokn6KEiGotlVInKt66QjBgXR2Nk9hIcez0LYt8W5pxD0lwTxC",
+			name:           "thi$$hoU_ld+*work",
+			AssociatedData: "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			Version:        -2000,
+			expStatus:      400,
+		},
+	}
+
+	t.Run("encrypt-decrypt handler", func(t *testing.T) {
+		fakeStateAuthority, _ := NewFakeSeedEngineAuthority()
+		mux := NewTransitEngineAPI(fakeStateAuthority, slog.Default())
+
+		for testName, entry := range mocks {
+			t.Run(testName, func(t *testing.T) {
+				var ciphertext, receivedPlaintext string
+				t.Run("encryption request handling", func(t *testing.T) {
+					require := require.New(t)
+					jsonBody, err := json.Marshal(entry)
+					require.NoError(err)
+
+					req := httptest.NewRequest(http.MethodPut, "/v1/transit/encrypt/"+entry.name, bytes.NewReader(jsonBody))
+					req.Header.Set("Content-Type", "application/json")
+
+					rec := httptest.NewRecorder()
+					mux.ServeHTTP(rec, req)
+					res := rec.Result()
+					require.Equal(entry.expStatus, res.StatusCode)
+					if res.StatusCode != http.StatusOK {
+						return
+					}
+
+					var encRespBody map[string]any
+					require.NoError(json.NewDecoder(res.Body).Decode(&encRespBody))
+
+					data, exist := encRespBody["data"].(map[string]any)
+					require.True(exist)
+					ciphertext, exist = data["ciphertext"].(string)
+					require.True(exist)
+				})
+
+				t.Run("decryption request handling", func(t *testing.T) {
+					require := require.New(t)
+					decryptReqBody, err := json.Marshal(
+						map[string]string{
+							"ciphertext":      ciphertext,
+							"associated_data": entry.AssociatedData,
+						})
+					require.NoError(err)
+
+					decryptReq := httptest.NewRequest(http.MethodPut, "/v1/transit/decrypt/"+entry.name, bytes.NewReader(decryptReqBody))
+					decryptReq.Header.Set("Content-Type", "application/json")
+
+					rec := httptest.NewRecorder()
+					mux.ServeHTTP(rec, decryptReq)
+					res := rec.Result()
+					require.Equal(entry.expStatus, res.StatusCode)
+					if res.StatusCode != http.StatusOK {
+						return
+					}
+
+					var decRespBody map[string]any
+					require.NoError(json.NewDecoder(res.Body).Decode(&decRespBody))
+
+					data, exist := decRespBody["data"].(map[string]any)
+					require.True(exist)
+					receivedPlaintext, exist = data["plaintext"].(string)
+					require.True(exist)
```

Comment:
These are not existence checks, but type cast checks. If the expected fields are not found, this will panic. You could avoid the type cast by assigning the expected type to begin with:

```
var decRespBody map[string]map[string]string
```